articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+
+A primary distinction is between running Q#:
+    - as a standalone application, where Q# is the only language involved and the program is invoked directly, or
+    - with an additional *host program*, written in Python or a .NET language (e.g. C# or F#), which then invokes the program and can further process returned results.
+
+To best understand the process and differences, we consider a simple Q# program and compare the ways it can be executed.
+
+## Basic Q# program
+
+It would be very surprising if you've made it this far without hearing the word "superposition."
+A very basic quantum program might consist of preparing a qubit in an equal superposition of states $\ket{0}$ and $\ket{1}$, measuring it, and returning the result, which will be randomly either one of the states with equal probability.
+Indeed, this process is at the core of the [quantum random number generator](xref:microsoft.quantum.quickstarts.qrng) quickstart.
+
+In Q#, this would be performed by the following code:
+
+```qsharp
+        using (q = Qubit()) {    // allocates qubit for use (automatically in |0>)
+            H(q);                // puts qubit in superposition of |0> and |1>
+            return MResetZ(q);   // measures qubit, returns result (and resets it to |0> before deallocation)
+        }
+```
+
+However, this code alone can't be executed by Q#.
+For that, it needs to make up the body of an [operation](xref:microsoft.quantum.guide.basics#q-operations-and-functions), which is then executed when called---either directly or by another operation. 
+Hence it takes the form
+```qsharp
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) {
+            H(q);
+            return MResetZ(q);
+        }
+    }
+```
+and we now have defined an operation, `MeasureSuperposition`, which takes no arguments and returns a value of type [Result](xref:microsoft.quantum.guide.types).
+More details on defining callables can be found at [Operations and functions](xref:microsoft.quantum.guide.operationsfunctions).
+For the sake of brevity we refer primarily to Q# *operations* on this page, however all of the concepts discussed work in the same manner for Q# *functions*, as well as any user-defined Q# types. 
+
+### Operation defined in a Q# file
+
+The operation is precisely what's called and run by Q#.
+However, it requires a few more additions to comprise a full `*.qs` Q# file.
+
+All Q# types and callables are defined within *namespaces*, which the compiler can then open individually 
+
+Firstly, the [`H`](xref:microsoft.quantum.intrinsic.h) and [`MResetZ`](xref:microsoft.quantum.measurement.mresetz) actually belong to the [Q# Standard Libraries](xref:microsoft.quantum.qsharplibintro).
+To be available for use, their respective *namespaces* need to be opened.
+All Q# callables and types (both those you define and those intrinsic to the language) are defined within namespaces.
```

Comment:
I'm not sure we want to get into this, but there is a slight edge case in that the Microsoft.Quantum.Core namespace is automatically opened so that callables like `Length` are always available.